mvlearn/ajive/ajive.py

```diff
@@ -0,0 +1,760 @@
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import numpy as np
+import warnings
+from scipy.sparse import issparse
+from copy import deepcopy
+from sklearn.externals.joblib import load, dump
+import pandas as pd
+from mvlearn.embed.base import BaseEmbed
+from mvlearn.utils.utils import check_Xs
+import warnings
+
+from .utils import svd_wrapper, centering
+from .wedin_bound import get_wedin_samples
+from .random_direction import sample_randdir
+from .diagnostic_plot import plot_joint_diagnostic
+from .pca import pca
+
+
+class ajive(object):
+    """
+    An implementation of Angle-based Joint and Individual Variation Explained.
+    This algorithm takes multiple input views with the same number of samples
+    and decomposes them into 3 distinct matrices representing:
+        - Individual variation of each particular view
+        - Joint variation shared by all views
+        - Noise
+
+    Parameters
+    ----------
+    init_signal_ranks: {list, dict}
+        The initial signal ranks.
+
+    joint_rank: {None, int}
+        Rank of the joint variation matrix. If None, will estimate the 
+        joint rank. Otherwise, will use provided joint rank.
+
+    indiv_ranks: {list, dict, None}
+        Ranks of individual variation matrices. If None, will estimate the 
+        individual ranks. Otherwise, will use provided individual ranks.
+
+    center: {bool, None}
+        Centers matrix. If None, will not center.
+
+    reconsider_joint_components: bool
+        Triggers _reconsider_joint_components function
+
+    wedin_percentile: int, default=5
+        Percentile used for wedin (lower) bound cutoff for squared 
+        singular values used to estimate joint rank.
+
+    n_wedin_samples: int, default=1000
+        Number of wedin bound samples to draw.
+
+    precomp_wedin_samples {None, dict of array-like, list of array-like}
+        Wedin samples that are precomputed for each view.
+
+    randdir_percentile: int, default=95
+        Percentile for random direction (lower) bound cutoff for squared
+        singular values used to estimate joint rank.
+
+    n_randdir_samples: int, default=1000
+        Number of random direction samples to draw.
+
+    precomp_randdir_samples {None,  array-like}
+        Precomputed random direction samples.
+
+    n_jobs: int, None
+        Number of jobs for parallel processing wedin samples and random
+        direction samples using sklearn.externals.joblib.Parallel.
+        If None, will not use parallel processing.
+
```

Comment:
I'm confused by the 'type' descriptions of each parameter. Can you put them all in the format like
int, default=100, rather than {bool, None}. Maybe I'm just confused

Reply:
I think something like list or dict, default=<DEFAULT> is more clear, but maybe that's just me